src/wireframes/shapes/neutral/tabs.ts

```diff
@@ -130,26 +133,48 @@ export class Tabs implements ShapePlugin {
         p.setStrokeColor(ctx.shape);
     }
 
-    private parseText(ctx: RenderContext, fontFamily: string, fontSize: number, strokeThickness: number) {
-        const isRight = ctx.shape.getAppearance(TAB_ALIGNMENT) === TAB_ALIGNMENT_RIGHT;
-        const key = `${ctx.shape.text}_${fontFamily}_${fontSize}_${strokeThickness}_${isRight}`;
+    private parseLines(shape: Shape) {
+        const key = `${ shape.text}`;
 
-        let result = ctx.shape.renderCache['PARSED'] as { key: string; parsed: Parsed };
+        let result = shape.renderCache['PARSED_TEXT'] as { key: string; parts: ParsedLines };
 
         if (!result || result.key !== key) {
-            const w = ctx.rect.width - 2 * PADDING;
-
-            const parsed: Parsed = [];
-
-            let x = 0;
+            const parts: ParsedLines = [];
 
-            for (let text of ctx.shape.text.split(',')) {
+            for (let text of shape.text.split(',')) {
                 const selected = text.endsWith('*');
 
                 if (selected) {
                     text = text.substring(0, text.length - 1).trim();
                 }
 
+                parts.push({ text, selected });
+            }
+
+            result = { key, parts };
+            shape.renderCache['PARSED_TEXT'] = result;
+        }
+
+        return result.parts;
+    }
+
+    private arrangeLines(ctx: RenderContext, fontFamily: string, fontSize: number, strokeThickness: number) {
+        const isRight = ctx.shape.getAppearance(TAB_ALIGNMENT) === TAB_ALIGNMENT_RIGHT;
+        const selectedFromAppearance = ctx.shape.getAppearance(TAB_SELECTED) as string | null;
+        console.log(selectedFromAppearance, selectedFromAppearance === null);
+        const key = `${ctx.shape.text}_${fontFamily}_${fontSize}_${strokeThickness}_${isRight}_${selectedFromAppearance}`;
+
+        let result = ctx.shape.renderCache['PARSED'] as { key: string; parsed: ArrangedLines };
+
+        if (!result || result.key !== key) {
+            const w = ctx.rect.width - 2 * PADDING;
+
+            const parsed: ArrangedLines = [];
+
+            let x = 0;
+
+            for (let { text, selected } of this.parseLines(ctx.shape)) {
+                selected ||= text == selectedFromAppearance;
```

Comment:
I am not sure if like that, this undos everything whenever I rename something. Would the index not be better?